src/EnergyPlus/EvaporativeCoolers.cc

```diff
@@ -1045,35 +1041,29 @@ void InitEvapCooler(EnergyPlusData &state, int const EvapCoolNum)
     // Uses the status flags to trigger events.
 
     // Using/Aliasing
-    bool DoSetPointTest = state.dataHVACGlobal->DoSetPointTest;
-
-    // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-    bool localSetPointCheck(false);
-
-    auto &EvapCond(state.dataEvapCoolers->EvapCond);
+    auto &evapCond = state.dataEvapCoolers->EvapCond(EvapCoolNum);
 
     // Check that setpoint is active
-    if (!state.dataGlobal->SysSizingCalc && state.dataEvapCoolers->MySetPointCheckFlag && DoSetPointTest) {
+    if (!state.dataGlobal->SysSizingCalc && state.dataEvapCoolers->MySetPointCheckFlag && state.dataHVACGlobal->DoSetPointTest) {
         for (int EvapUnitNum = 1; EvapUnitNum <= state.dataEvapCoolers->NumEvapCool; ++EvapUnitNum) {
 
             // only check evap coolers that are supposed to have a control node
-            if ((EvapCond(EvapCoolNum).evapCoolerType != EvapCoolerType::IndirectRDDSpecial) &&
-                (EvapCond(EvapCoolNum).evapCoolerType != EvapCoolerType::DirectResearchSpecial))
+            if ((evapCond.evapCoolerType != EvapCoolerType::IndirectRDDSpecial) && (evapCond.evapCoolerType != EvapCoolerType::DirectResearchSpecial))
                 continue;
 
-            int ControlNode = EvapCond(EvapUnitNum).EvapControlNodeNum;
+            int ControlNode = state.dataEvapCoolers->EvapCond(EvapUnitNum).EvapControlNodeNum;
             if (ControlNode > 0) {
                 if (state.dataLoopNodes->Node(ControlNode).TempSetPoint == DataLoopNode::SensedNodeFlagValue) {
                     if (!state.dataGlobal->AnyEnergyManagementSystemInModel) {
-                        ShowSevereError(state, format("Missing temperature setpoint for Evap Cooler unit {}", EvapCond(EvapCoolNum).Name));
+                        ShowSevereError(state, format("Missing temperature setpoint for Evap Cooler unit {}", evapCond.Name));
                         ShowContinueError(state, " use a Setpoint Manager to establish a setpoint at the unit control node.");
                     } else {
-                        localSetPointCheck = false;
+                        bool localSetPointCheck = false;
                         CheckIfNodeSetPointManagedByEMS(state, ControlNode, EMSManager::SPControlType::TemperatureSetPoint, localSetPointCheck);
                         state.dataLoopNodes->NodeSetpointCheck(ControlNode).needsSetpointChecking = false;
                         // Let it slide apparently
                         if (localSetPointCheck) {
-                            ShowSevereError(state, format("Missing temperature setpoint for Evap Cooler unit {}", EvapCond(EvapCoolNum).Name));
+                            ShowSevereError(state, format("Missing temperature setpoint for Evap Cooler unit {}", evapCond.Name));
```

Comment:
These 2 warnings are also wrong if using EvapCond(EvapCoolNum) instead of EvapCond(EvapUnitNum). The control node being tested is `ControlNode = EvapCond(EvapUnitNum).EvapControlNodeNum;`